### PR TITLE
Improve `make cluster-deploy` and `make cluster-clean`

### PR DESCRIPTION
**Pull request by davidvossel**
The setup time in `make cluster-deploy` has been improved by waiting to post the ocs subscription until after the catalog source is available. This improves the response time for how quickly the olm process the subscription.

The `make cluster-clean` script now completely removes everything related to `make cluster-up` by clearing the cephcluster finalizers and forcing all resources to be terminated.

### CODE REVIEW

**Comment by davidvossel** [comment]
these improvements should cut around 10 minutes from the setup time, which will improve the responsiveness of our PR e2e tests as well.

File: hack/cluster-clean.sh [hunk 1]
```diff
@@ -4,12 +4,28 @@ set -e
...
+oc -n openshift-storage delete storagecluster,cephcluster --all
+
 # even though we alter the deploy-with-olm.yaml in cluster-deploy,
 # the object names and namespaces remain the same, which allows us
 # to clean the cluster using the original manifest.
+echo "Deleting ocs-operator manifests"
 oc delete --ignore-not-found -f deploy/deploy-with-olm.yaml
 
+echo "Removing pods, pvcs, and pvs"
+oc -n openshift-storage delete pods,pvc,pv --all
```
Comment (jarrpa) [inline hack/cluster-clean.sh]: `oc delete --ignore-not-found -f deploy/deploy-with-olm.yaml` removes the `openshift-storage` namespace, right? Shouldn't this go before that?

Comment (davidvossel) [reply hack/cluster-clean.sh]: yeah, good point. I'm revising this...  it technically works as is, but not for obvious reasons.

**Comment by davidvossel** [comment]
> Are you just being extra thorough in deleting the different kinds of resources that we generate? Wouldn't it be more efficient to just delete the namespace and then any non-namespaced items?

Some objects need to be deleted in a specific order to avoid having the Namespace wedged in a "terminating" state forever

The biggest complication here is that noobaa-core is consuming PVC/PVs created by the ceph cluster. That means we have to tear things down in a way that ensure we remove noobaa-core before tearing down the underlying mechanisms that supply the noobaa-core's storage. we also have to be careful to remove operands that result in the creation of the noobaa-core statefulset before removing the noobaa-core, or else noobaa-core will get immediately re-created after we delete it. 

So there's some "unwinding" that has to occur before we can safely delete the Namespace.

**Comment by jarrpa** [comment]
@davidvossel Yeah, shortly after I said that I ran into that very thing this weekend 😂

**Comment by rohantmp** [comment]
@davidvossel  @jarrpa Sorry I missed this before merge.

Doesn't deleting storageclusterinitializations trigger other resources to be deleted.

**Comment by davidvossel** [comment]
> Doesn't deleting storageclusterinitializations trigger other resources to be deleted?

the storageclusterinitializations is created by the StorageCluster object, so i think we'd end up potentially just having the reconcile loop re-creating the storageclusterinitialization right after deleting it.

**Pull request merged**